streams/src/main/java/org/apache/kafka/streams/processor/TaskId.java

```diff
@@ -59,8 +58,8 @@ public TaskId(final int topicGroupId, final int partition, final String namedTop
         }
     }
 
-    public Optional<String> namedTopology() {
-        return namedTopology == null ? Optional.empty() : Optional.of(namedTopology);
+    public String namedTopology() {
```

Comment:
Are we going to to just make all subtopologies named ?

Reply:
I have considered a bit whether to do that, and just make the "normal" (unnamed) case like a specific case of the named topology where the name is just the app id or empty string or something, but I think that ultimately only complicates things even more, so I decided against it. For example we'd need to check whether the named topology was equal to `app.id`/`""` before placing task dirs into named subdirectories, inserting the name into internal topics, etc. It's easier to just check whether it exists, that way it's pretty obvious if you miss something, vs a subtle bug that eg changed the internal topic names by one extra `-` 

But once we circle back on making this a public API I think we can clean things up a lot, at which time it may make more sense to consolidate things in that way

Reply:
Once we make this a public API would make them all named but I am okay with waiting